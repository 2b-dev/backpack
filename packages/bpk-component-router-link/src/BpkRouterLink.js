import React, { PropTypes } from 'react'
import { Link, PropTypes as RouterPropTypes } from 'react-router'

import './bpk-router-link.scss'

const BpkRouterLink = ({ children, to, onClick }) => (
  <Link
    className='bpk-router-link'
    activeClassName='bpk-router-link--active'
    children={children}
    to={to}
    onClick={onClick}
  />
)

BpkRouterLink.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
  to: PropTypes.oneOfType([
    PropTypes.shape(RouterPropTypes.locationShape),
    PropTypes.string
  ]).isRequired,
  onClick: PropTypes.func
}

BpkRouterLink.defaultProps = {
  onClick: null
}

export default BpkRouterLink
