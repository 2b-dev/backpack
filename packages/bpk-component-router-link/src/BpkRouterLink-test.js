import React from 'react';
import renderer from 'react-test-renderer';
import BpkRouterLink from './BpkRouterLink';

describe('BpkRouterLink', () => {
  it('should render correctly with a "to" attribute', () => {
    const tree = renderer.create(<BpkRouterLink to="#">Link</BpkRouterLink>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly with a "className" attribute', () => {
    const tree = renderer.create(
      <BpkRouterLink to="#" className="my-custom-class-1 my-custom-class-2">Link</BpkRouterLink>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly with arbitrary attributes', () => {
    const tree = renderer.create(<BpkRouterLink to="#" target="_blank">Link</BpkRouterLink>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
