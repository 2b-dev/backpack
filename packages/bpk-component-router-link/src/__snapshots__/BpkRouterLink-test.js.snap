exports[`BpkRouterLink should render correctly with a "blank" attribute 1`] = `
<a
  className="bpk-router-link"
  onClick={[Function bound handleClick]}
  style={Object {}}>
  Link (new window)
</a>
`;

exports[`BpkRouterLink should render correctly with a "to" attribute 1`] = `
<a
  className="bpk-router-link"
  onClick={[Function bound handleClick]}
  style={Object {}}>
  Link
</a>
`;
